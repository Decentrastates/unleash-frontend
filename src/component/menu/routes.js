import CreateFeatureToggle from '../../page/features/create';
import CopyFeatureToggle from '../../page/features/copy';
import ViewFeatureToggle from '../../page/features/show';
import Features from '../../page/features';
import CreateStrategies from '../../page/strategies/create';
import StrategyView from '../../page/strategies/show';
import Strategies from '../../page/strategies';
import HistoryPage from '../../page/history';
import HistoryTogglePage from '../../page/history/toggle';
import ShowArchive from '../../page/archive/show';
import Archive from '../../page/archive';
import Applications from '../../page/applications';
import ApplicationView from '../../page/applications/view';
import LogoutFeatures from '../../page/user/logout';
import ExperimentList from '../../page/experiment';
import ExperimentCreate from '../../page/experiment/create';
import ExperimentShow from '../../page/experiment/show';

export const routes = [
    // Features
    { path: '/features/create', parent: '/features', title: 'Create', component: CreateFeatureToggle },
    { path: '/features/copy/:copyToggle', parent: '/features', title: 'Copy', component: CopyFeatureToggle },
    { path: '/features/:activeTab/:name', parent: '/features', title: ':name', component: ViewFeatureToggle },
    { path: '/features', title: 'Feature Toggles', icon: 'list', component: Features },

    // Strategies
    { path: '/strategies/create', title: 'Create', parent: '/strategies', component: CreateStrategies },
    {
        path: '/strategies/:activeTab/:strategyName',
        title: ':strategyName',
        parent: '/strategies',
        component: StrategyView,
    },
    { path: '/strategies', title: 'Strategies', icon: 'extension', component: Strategies },

    // History
    { path: '/history/:toggleName', title: ':toggleName', parent: '/history', component: HistoryTogglePage },
    { path: '/history', title: 'Event History', icon: 'history', component: HistoryPage },

    // Archive
    { path: '/archive/:activeTab/:name', title: ':name', parent: '/archive', component: ShowArchive },
    { path: '/archive', title: 'Archived Toggles', icon: 'archive', component: Archive },

    // Applications
    { path: '/applications/:name', title: ':name', parent: '/applications', component: ApplicationView },
    { path: '/applications', title: 'Applications', icon: 'apps', component: Applications },

    { path: '/logout', title: 'Sign out', icon: 'exit_to_app', component: LogoutFeatures },

    //
    { path: '/experiments/create', parent: '/experiments', title: 'Create', component: ExperimentCreate },
    { path: '/experiments/:experimentName', parent: '/experiments', title: ':experimentName', component: ExperimentShow },
    { path: '/experiments', title: 'Experiments', icon: 'emoji_flags', component: ExperimentList },
];

export const getRoute = path => routes.find(route => route.path === path);

export const baseRoutes = routes.filter(route => !route.parent);
