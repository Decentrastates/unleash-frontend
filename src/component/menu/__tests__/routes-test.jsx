import { baseRoutes, getRoute } from '../routes';

test('returns all baseRoutes', () => {
    expect(baseRoutes).toHaveLength(38);
    expect(baseRoutes).toMatchSnapshot();
});

test('getRoute() returns named route', () => {
    const featuresRoute = getRoute('/features');
    expect(featuresRoute.path).toEqual('/features');
});
