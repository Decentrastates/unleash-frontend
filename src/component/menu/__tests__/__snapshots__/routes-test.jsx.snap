// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns all baseRoutes 1`] = `
Array [
  Object {
    "component": [Function],
    "layout": "main",
    "menu": Object {
      "mobile": true,
    },
    "path": "/features",
    "title": "Feature Toggles",
    "type": "protected",
  },
  Object {
    "component": [Function],
    "layout": "main",
    "menu": Object {
      "advanced": true,
      "mobile": true,
    },
    "path": "/strategies",
    "title": "Strategies",
    "type": "protected",
  },
  Object {
    "component": [Function],
    "layout": "main",
    "menu": Object {
      "adminSettings": true,
    },
    "path": "/history",
    "title": "Event History",
    "type": "protected",
  },
  Object {
    "component": [Function],
    "layout": "main",
    "menu": Object {},
    "path": "/archive",
    "title": "Archived Toggles",
    "type": "protected",
  },
  Object {
    "component": [Function],
    "layout": "main",
    "menu": Object {
      "advanced": true,
      "mobile": true,
    },
    "path": "/applications",
    "title": "Applications",
    "type": "protected",
  },
  Object {
    "component": [Function],
    "flag": "C",
    "layout": "main",
    "menu": Object {
      "advanced": true,
      "mobile": true,
    },
    "path": "/context",
    "title": "Context Fields",
    "type": "protected",
  },
  Object {
    "component": [Function],
    "flag": "P",
    "layout": "main",
    "menu": Object {
      "mobile": true,
    },
    "path": "/projects",
    "title": "Projects",
    "type": "protected",
  },
  Object {
    "component": [Function],
    "layout": "main",
    "menu": Object {
      "advanced": true,
      "mobile": true,
    },
    "path": "/tag-types",
    "title": "Tag types",
    "type": "protected",
  },
  Object {
    "component": [Function],
    "hidden": false,
    "layout": "main",
    "menu": Object {
      "advanced": true,
      "mobile": true,
    },
    "path": "/addons",
    "title": "Addons",
    "type": "protected",
  },
  Object {
    "component": [Function],
    "layout": "main",
    "menu": Object {
      "advanced": true,
      "mobile": true,
    },
    "path": "/reporting",
    "title": "Reporting",
    "type": "protected",
  },
  Object {
    "component": [Function],
    "hidden": false,
    "layout": "main",
    "menu": Object {},
    "path": "/admin",
    "title": "Admin",
    "type": "protected",
  },
]
`;
